src/schemas/TimesheetSchema.tsx

```diff
@@ -1,12 +1,31 @@
 import { z } from "zod";
+import {RowSchema} from './RowSchema'; 
+
+// The status is either undefined, for not being at that stage yet, or 
+// contains the date and author of approving this submission 
+export const StatusEntryType = z.union(
+  [
+  z.object({
+    Date: z.number(), 
+    Author: z.string()
+  }), 
+  z.undefined()]); 
+
+// Status type contains the four stages of the pipeline we have defined 
+export const StatusType = z.object({
+  Employee: StatusEntryType, 
+  Manager: StatusEntryType,
+  Admin: StatusEntryType, 
+  Accepted: StatusEntryType 
+});
 
 export const TimeSheetSchema = z.object({
```

Comment:
I also have an array of Notes here (comments/reports) called WeekNotes that represents notes left for an entire week